File: webapp-fchat/src/app/stores/message/message.state.ts
import { Conversation, Message } from "./message.interface";

export interface MessageState{
    allConversations: Conversation[] | null  | undefined,
    currentConversationId: string | null;
    currentConversation:  Conversation | null  | undefined
    isLoading: boolean,
    error: string | null;
}


export const initialMessageState: MessageState = {
    allConversations: null,
    currentConversationId: null,
    currentConversation: null,
    isLoading: false,
    error: '',
}

File: mobile_fchat/lib/state/blocs/message/message.state.dart
import 'package:mobile_fchat/state/blocs/user/user.state.dart';
import 'package:mobile_fchat/state/models/conversation.dart';

class MessageState {
  final Status? status;
  final List<Conversation>? allConversations;
  final List<Conversation>? allConversationsFiltered;
  final Conversation? currentConversation;
  final dynamic errors;

  MessageState({this.status, this.errors, this.allConversations, this.currentConversation, this.allConversationsFiltered});

  MessageState copyWith({
    Status? status,
    dynamic errors,
    Conversation? currentConversation,
    List<Conversation>? allConversations,
    List<Conversation>? allConversationsFiltered,
  }) {
    return MessageState(
      status: status,
      errors: errors,
      currentConversation: currentConversation ?? this.currentConversation,
      allConversations: allConversations ?? this.allConversations,
      allConversationsFiltered: allConversationsFiltered ?? this.allConversationsFiltered,
    );
  }
}
